File: Patterns/10. Hourglass Patterns/94. Hourglass Decreasing Numbers/hdn.cpp
#include <iostream>
#include <algorithm>
using namespace std;
int main(){
    int n,row;
    cout<<"Enter a Number : ";
    cin>>n;

    for(int i=1; i<=2*n-1; i++){

        row = min(i,2*n-i);

        // print spaces
        for(int j=1; j<=row-1; j++){
            cout<<"  ";
        }
        // print first half
        for(int j=n; j>=row; j--){
            cout<<j<<" ";
        }
        // print second half
        for(int j=row+1; j<=n; j++){
            cout<<j<<" ";
        }
        cout<<'\n';

    }
}

/*
Output

5 4 3 2 1 2 3 4 5 
  5 4 3 2 3 4 5 
    5 4 3 4 5 
      5 4 5 
        5 
      5 4 5 
    5 4 3 4 5 
  5 4 3 2 3 4 5 
5 4 3 2 1 2 3 4 5  

*/

File: Patterns/10. Hourglass Patterns/94. Hourglass Decreasing Numbers/hdn.py
n = int(input("Enter a Number : "))

for i in range(1,2*n):

    row = min(i,2*n-i)
    
    # print spaces
    for j in range(1,row):
        print(end='  ')
    # print first half
    for j in range(n,row-1,-1):
        print(j, end=' ')
    # print second half
    for j in range(row+1,n+1):
        print(j, end=' ')
    print()


'''
Output:

5 4 3 2 1 2 3 4 5 
  5 4 3 2 3 4 5 
    5 4 3 4 5 
      5 4 5 
        5 
      5 4 5 
    5 4 3 4 5 
  5 4 3 2 3 4 5 
5 4 3 2 1 2 3 4 5  


'''